src/distinguished_name.cpp

```diff
@@ -84,22 +92,32 @@ DistinguishedName DistinguishedName::fromX509Name(X509_NAME *x509)
     return builder.build();
 }
 
-auto _createTuple(const DistinguishedName &dn)
+auto _createTuple(const DistinguishedName& dn)
 {
-    return std::tie(dn.commonName(),
+    return std::make_tuple(dn.commonName(),
                     dn.countryName(),
-                    dn.localityName(),
-                    dn.stateOrProvinceName(),
-                    dn.organizationalUnitName(),
-                    dn.organizationName(),
-                    dn.pkcs9EmailAddress(),
-                    dn.serialNumber(),
-                    dn.userId());
+                     dn.localityName(),
+                     dn.stateOrProvinceName(),
+                     dn.organizationalUnitName(),
+                     dn.organizationName(),
+                     dn.pkcs9EmailAddress(),
+                     dn.serialNumber(),
+                     dn.userId(),
+                     dn.givenName());
 }
 
 bool DistinguishedName::operator==(const DistinguishedName &other) const
 {
     return _createTuple(*this) == _createTuple(other);
 }
 
+std::string DistinguishedName::_getAttributeByNIDAsString(const ASN1_NID id) const
+{
+    const auto it = std::find_if(_subjectAttributes.rbegin(), _subjectAttributes.rend(), [&id] (const auto& attribute) { return attribute.id == id; } );
```

Comment:
You've done this multiple times now, so I'd like to point out that there's very little point in declaring a ASN1_NID const or including it by reference in a lambda.

An ASN1_NID is an enum, which is probably just an int. For the same reason that you usually don't see `const int` function parameters (they are copied anyway), you don't really need a `const ASN1_NID`. Additionally, including a reference in the lambda is actually more expensive, because you copy a pointer, which is likely larger than an integer.

This is definitely a nitpick, though, because it really don't make any difference whatsoever at runtime.

Reply:
In my opinion its good practice to make a read only input parameter also readonly by design. The compiler is then complaining in case someone modifies it within the function. Sure this function is simple but there might be more complex once. Further down the lines someone sets the value to something else accidently -> boom. Sure we can the reference passing can be removed if you really dont like it.

Reply:
I don't have a very strong opinion about this. I figured I'd mention it since you actually do change it from the previous state in a couple of places.